ddev/src/ddev/cli/validate/ci.py

```diff
@@ -0,0 +1,89 @@
+
+    app: Application = ctx.obj
+    test_workflow = (
+        './.github/workflows/test-target.yml'
+        if app.repo.name == 'core'
+        else 'DataDog/integrations-core/.github/workflows/test-target.yml@master'
+    )
+    jobs_workflow_path = app.repo.path / '.github' / 'workflows' / 'test-all.yml'
+    original_jobs_workflow = jobs_workflow_path.read_text() if jobs_workflow_path.is_file() else ''
+
+    jobs = {}
+    for data in construct_job_matrix(app.repo.path, get_all_targets(app.repo.path)):
+        python_restriction = data.get('python-support', '')
+        config = {
+            'job-name': data['name'],
+            'target': data['target'],
+            'platform': data['platform'],
+            'runner': json.dumps(data['runner'], separators=(',', ':')),
+            'repo': '${{ inputs.repo }}',
+            # Options
+            'python-version': '${{ inputs.python-version }}',
+            'standard': '${{ inputs.standard }}',
+            'latest': '${{ inputs.latest }}',
+            'minimum-base-package': '${{ inputs.minimum-base-package }}',
+            'agent-image': '${{ inputs.agent-image }}',
+            'agent-image-py2': '${{ inputs.agent-image-py2 }}',
+            'agent-image-windows': '${{ inputs.agent-image-windows }}',
+            'agent-image-windows-py2': '${{ inputs.agent-image-windows-py2 }}',
+            'test-py2': '2' in python_restriction if python_restriction else '${{ inputs.test-py2 }}',
+            'test-py3': '3' in python_restriction if python_restriction else '${{ inputs.test-py3 }}',
+            'setup-env-vars': '${{ inputs.setup-env-vars }}',
+        }
+
+        # Prevent redundant job hierarchy names at the bottom of pull requests and also satisfy the naming requirements:
+        # https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#jobsjob_id
+        job_id = hashlib.sha256(config['job-name'].encode('utf-8')).hexdigest()[:7]
+        job_id = f'j{job_id}'
+
+        jobs[job_id] = {'uses': test_workflow, 'with': config, 'secrets': 'inherit'}
+
+    jobs_component = yaml.safe_dump({'jobs': jobs}, default_flow_style=False, sort_keys=False)
+
+    # Enforce proper string types
+    for field in (
+        'repo',
+        'python-version',
+        'setup-env-vars',
+        'agent-image',
+        'agent-image-py2',
+        'agent-image-windows',
+        'agent-image-windows-py2',
+    ):
+        jobs_component = jobs_component.replace(f'${{{{ inputs.{field} }}}}', f'"${{{{ inputs.{field} }}}}"')
+
+    manual_component = original_jobs_workflow.split('jobs:')[0].strip()
+    expected_jobs_workflow = f'{manual_component}\n\n{jobs_component}'
+
+    if original_jobs_workflow != expected_jobs_workflow:
+        if sync:
+            jobs_workflow_path.write_text(expected_jobs_workflow)
+        else:
+            app.abort('CI configuration is not in sync, try again with the `--sync` flag')
+
+    ctx.invoke(legacy_validation, fix=sync)
```

Comment:
Do I understand correctly that this will be called independent of whether `original_jobs_workflow != expected_jobs_workflow` and *after* we write out `expected_jobs_workflow`?

Reply:
This won't be called if the former fails

Reply:
ah, right, cuz in CI we won't pass `--sync`. All clear now, thanks!